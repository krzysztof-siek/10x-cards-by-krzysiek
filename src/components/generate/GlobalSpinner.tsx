export const GlobalSpinner = () => {
  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black/50 z-50">
      <div className="bg-white p-6 rounded-lg shadow-lg flex flex-col items-center">
        <div className="h-10 w-10 border-4 border-t-blue-500 border-r-transparent border-b-blue-500 border-l-transparent rounded-full animate-spin mb-4"></div>
        <p className="text-gray-700 font-medium">Generating flashcards...</p>
        <p className="text-gray-500 text-sm mt-1">This may take a moment</p>
      </div>
    </div>
  );
}; 