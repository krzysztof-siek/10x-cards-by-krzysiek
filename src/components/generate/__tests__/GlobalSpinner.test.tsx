import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { GlobalSpinner } from "../GlobalSpinner";

describe("GlobalSpinner", () => {
  it("renders with loading text", () => {
    render(<GlobalSpinner />);

    expect(screen.getByText("Generowanie fiszek...")).toBeInTheDocument();
    expect(screen.getByText("To może chwilę potrwać")).toBeInTheDocument();
  });

  it("renders a spinner animation element", () => {
    render(<GlobalSpinner />);

    const spinnerElement = document.querySelector(".animate-spin");
    expect(spinnerElement).toBeInTheDocument();
  });

  it("has a semi-transparent background overlay", () => {
    render(<GlobalSpinner />);

    const overlay = document.querySelector(".bg-black\\/50");
    expect(overlay).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<GlobalSpinner />);
    expect(container).toMatchSnapshot();
  });
});
