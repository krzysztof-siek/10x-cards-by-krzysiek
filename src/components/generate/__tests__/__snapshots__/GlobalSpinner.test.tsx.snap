// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GlobalSpinner > should match snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 flex items-center justify-center bg-black/50 z-50"
  >
    <div
      class="bg-white p-6 rounded-lg shadow-lg flex flex-col items-center"
    >
      <div
        class="h-10 w-10 border-4 border-t-blue-500 border-r-transparent border-b-blue-500 border-l-transparent rounded-full animate-spin mb-4"
      />
      <p
        class="text-gray-700 font-medium"
      >
        Generowanie fiszek...
      </p>
      <p
        class="text-gray-500 text-sm mt-1"
      >
        To może chwilę potrwać
      </p>
    </div>
  </div>
</div>
`;
